config/tectonic-ingress/config.go

```diff
@@ -0,0 +1,32 @@
+package tectonicingress
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// Kind is the TypeMeta.Kind for the OperatorConfig.
+	Kind = "TectonicIngressControllerConfig"
+	// APIVersion is the TypeMeta.APIVersion for the OperatorConfig.
+	APIVersion = "v1"
+)
+
+// IngressType selects the ingress controller profile to install
+//
+type IngressType string
+
+const (
+	// IngressNginxNodePort is the nginx ingress controller with a NodePort service
```

Comment:
Do we want to provide knobs at this low-level, or can we make intelligent decisions based on something higher-level (such as the cloud provider?)

We'd like to take as much path-dependent decisionmaking out of the installer side as possible.

Reply:
It depends. I was anticipating additional ingress controllers as best practices evolve. So, right now, there's a simple choice based on platform, but that may not continue to be the case.

Also, this would mean reading a configmap outside-namespace. Not a problem, but it may break the mental model.

Reply:
I'd rather have the operator make the choice rather than someone on the outside. Ideally we want the installer side to pass through specific config values (cluster CIDR, cloud provider, etc.) and have the operator make those decisions. If platform is all that is needed for now I'd rather go with that.

Not sure I follow the configmap-outside-namespace part? Each namespace has a cluster-config configmap, we would put your config inside the same namespace as your operator. The installer can template in whichever values you want.